src/tools/pmempool/rm.c

```diff
@@ -242,35 +246,89 @@ static int
 rm_poolset_cb(struct part_file *pf, void *arg)
 {
 	int *error = (int *)arg;
-	int ret;
+	int ret = 0;
 	if (pf->is_remote) {
 		ret = remove_remote(pf->remote->node_addr,
 					pf->remote->pool_desc);
+	} else {
+		const char *part_file = pf->part->path;
+		ret = rm_file(part_file);
+	}
+
+	if (ret)
+		*error = ret;
+
+	return 0;
+}
+
+#ifdef USE_RPMEM
+/* XXX: replace with real size of the poolset */
+#define POOL_SIZE (32 * 1024 * 1024)
+
+static void *
+alloc_memory()
+{
+	/* allocate a page size aligned local memory pool */
+	void *mem;
+	int ret = posix_memalign(&mem, Pagesize, POOL_SIZE);
+	if (ret) {
+		fprintf(stderr, "posix_memalign: %s\n", strerror(ret));
+		exit(1);
+	}
+
+	return mem;
+}
+#endif
+
+/*
+ * poolset_parts_exist -- (internal) checks if given pool exists
+ */
+static int
+poolset_parts_exist(struct part_file *pf, void *arg)
+{
+	int *error = (int *)arg;
+	int ret = 0;
+	if (pf->is_remote) {
+#ifdef USE_RPMEM
+		struct rpmem_pool_attr pool_attr;
+		unsigned nlanes = 1;
+
+		void *pool = alloc_memory();
```

Comment:
This is dangerous (zeroes in the local pool may confuse some code) and may actually require an allocation of a lot of DRAM (as rdma registration may require pinned pages).